tests/densify/test_interface_patchmatchnet.py

```diff
@@ -0,0 +1,160 @@
+
+    def setUp(self) -> None:
+        """Set up the image dictionary and gtsfm result for the test."""
+        super().setUp()
+
+        # set the number of images as the default number
+        self._num_images = DEFAULT_NUM_IMAGES
+
+        # set up the default image dictionary
+        self._img_dict = DEFAULT_DUMMY_IMAGE_DICT
+
+        # initialize an empty sfm result in GtsfmData class
+        self._sfm_result = GtsfmData(self._num_images)
+        # add default cameras to the sfm result
+        self.add_default_cameras()
+        # add default tracks to the sfm result
+        self.add_default_tracks()
+        # build PatchmatchNet dataset from input images and the sfm result
+        self._dataset_patchmatchnet = PatchmatchNetData(self._img_dict, self._sfm_result)
+
+    def add_default_cameras(self) -> None:
+        """Assume all default cameras are valid cameras, add them into sfm_result"""
+        self._num_valid_cameras = DEFAULT_NUM_CAMERAS
+        for i in range(DEFAULT_NUM_CAMERAS):
+            self._sfm_result.add_camera(i, DEFAULT_CAMERAS[i])
+
+    def add_default_tracks(self) -> None:
+        """Calculate the measurements under each camera for all track points, then insert the tracks into sfm result"""
+
+        self._num_tracks = DEFAULT_NUM_TRACKS
+        for j in range(self._num_tracks):
+            world_x = DEFAULT_TRACK_POINTS[j]
+            track_to_add = SfmTrack(world_x)
+
+            for i in range(self._num_valid_cameras):
+                measurement = self._sfm_result.get_camera(i).project(world_x)
+                track_to_add.add_measurement(idx=i, m=measurement)
+            self._sfm_result.add_track(track_to_add)
+
+    def test_dataset_length(self) -> None:
+        """Test whether the dataset length is equal to the number of valid cameras, because every valid camera will be
+        regarded as reference view once."""
+        self.assertEqual(len(self._dataset_patchmatchnet), self._num_valid_cameras)
+
+    def test_select_src_views(self) -> None:
+        """Test whether the (ref_view, src_view) pairs are selected correctly."""
+        self.assertTrue(np.all(self._dataset_patchmatchnet._pairs == np.array([[1, 2], [0, 2], [0, 1]])))
+
+    def test_depth_ranges(self) -> None:
+        """Test whether the depth ranges for each camera are calculated correctly."""
+        # extract depth values for every track point in each camera's frame
+        depths = [
+            [
+                self._sfm_result.get_camera(i).pose().transformTo(DEFAULT_TRACK_POINTS[j])[-1]
+                for j in range(DEFAULT_NUM_TRACKS)
+            ]
+            for i in range(DEFAULT_NUM_CAMERAS)
+        ]
+        # calculate the depth range for each camera, using the pre-defined minimum and maximum depth percentile
+        depth_ranges = [
+            [np.percentile(depths[i], MIN_DEPTH_PERCENTILE), np.percentile(depths[i], MAX_DEPTH_PERCENTILE)]
```

Comment:
Thanks for adding the unit tests, Ren. It looks like the unit test is repeating the code used in the original module. Could we use actual numbers here instead?

If the 3d points are at (0.5, 0.5, 0), (-0.1, -1.0, 0.1) and the camera poses are 
    Pose3(Rot3(np.array([[0, 0, -1], [0, 1, 0], [1, 0, 0]])), np.array([1, 0, 0])),
    Pose3(Rot3(np.array([[-1, 0, 0], [0, 1, 0], [0, 0, -1]])), np.array([0, 0, 1]))

then what would be the actual floating point depth ranges? Can we add a unit test where there are 2 outliers as well (one close and one far), and suppose the same inlier is repeated 98 times. Does the percentile computation work?